### PR TITLE
Fix StorageProfile incomplete metric test flakiness

### PR DESCRIPTION
**Pull request by arnongilboa**
Signed-off-by: Arnon Gilboa <agilboa@redhat.com>

**What this PR does / why we need it**:
Flakiness seems to be introduced by #2546 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: tests/operator_test.go [hunk 1]
```diff
@@ -981,18 +981,24 @@ var _ = Describe("ALL Operator tests", func() {
 					Skip("This test depends on prometheus infra being available")
 				}
 
-				numAddedStorageClasses = 2
 				defaultStorageClass := utils.GetDefaultStorageClass(f.K8sClient)
 				defaultStorageClassProfile := &cdiv1.StorageProfile{}
 				err := f.CrClient.Get(context.TODO(), types.NamespacedName{Name: defaultStorageClass.Name}, defaultStorageClassProfile)
 				Expect(err).ToNot(HaveOccurred())
+
+				originalMetricVal := 0
```
Comment (akalenyu) [inline tests/operator_test.go:989]: I think originally the metric values is not 0 because we have the "local" storage class

Comment (awels) [reply tests/operator_test.go:989]: But local should not trigger the warning anymore since we added it to the storage profile.

Comment (akalenyu) [reply tests/operator_test.go:989]: did we? where?

Comment (arnongilboa) [reply tests/operator_test.go:989]: Due to PR #2546 playing with CDI CR, in this test case it takes `StorageProfileReconciler` a second or two to get to its `IncompleteProfileGauge.Set(>0)`.

**Review by akalenyu — COMMENTED** [review]
what was the flake here? IIUC the other PR introduced a test that did not mess around with storage classes at all

**Comment by arnongilboa** [comment]
> what was the flake here? IIUC the other PR introduced a test that did not mess around with storage classes at all

see my comment above

File: tests/operator_test.go (hunk 1 above)
Comment (awels) [reply tests/operator_test.go:989]: Interesting we only did it for LSO not U/S local storage.

Comment (akalenyu) [reply tests/operator_test.go:989]: yeah because you still need to alert when someone is rolling their own static PVs

**Pull request merged**